Week 04/Lecture 08/Assignment 3/src/main/java/com/lecture8/assignment3/service/EmployeeServiceImpl.java

```diff
@@ -0,0 +1,61 @@
+package com.lecture8.assignment3.service;
+
+import java.util.List;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.lecture8.assignment3.entity.Employee;
+import com.lecture8.assignment3.repository.EmployeeRepository;
+
+@Service
+@Transactional
```

Comment:
How about @Transactional put in each method? Pls explain. Should figure out what is name of Transaction manager.

Reply:
By annotating individual methods with @Transactional, we gain fine-grained control over transaction management. Each method operates within its own transactional context. 

This ensures that if a failure occurs within a method, only that method's transaction is rolled back, and other transactions remain unaffected. 

If an exception occurs within a method, the transaction is automatically rolled back, preventing partial updates to the database and maintaining data integrity.

I have update the code to applied @Transactional annotation to each method individually. I have update @Transactional(readOnly = true) is used on findAll and findEmployeeById methods, which do not modify the database, ensuring these methods are executed in a read-only transaction.

Regarding the transaction manager, in a Spring application that uses JDBC, the default transaction manager is DataSourceTransactionManager. This transaction manager handles the transaction boundaries and ensures proper transaction management for JDBC operations

Reply:
good explaination. but, as I mentioned you the important thing is you should figure out Transaction manager name in @Transactional anotation, it should be better, clearer (in case we have multiple datasource, it dont know what is datasource using in transaction)